### ISSUE TITLE
False negative for `SlotReusedDetector` when slot passed as lambda without being invoked

### ISSUE THREAD

**Issue by alexvanyo** [slackhq/compose-lints#404]
`SlotReusedDetector` currently won't detect problematic code that looks like the following:

```kotlin
@Composable
fun Something(
  flag: Boolean,
  modifier: Modifier = Modifier,
  slot: @Composable () -> Unit,
) {
  Box(modifier) {
    if (flag) {
      MaterialTheme(slot)
    } else {
      MaterialTheme(slot)
    }
  }
}
```

This will lose the internal state of `slot` when the `flag` changes value, but won't currently be caught by `SlotReusedDetector`.

**Comment by alexvanyo** [comment]
It might be tricky to solve this false negative without introducing false positives for correct manipulations of `@Composable` lambdas.

One reasonable assumption that might be useful to make is that if another `@Composable` function takes a `@Composable` slot parameter itself (in this case, `MaterialTheme`), then it intends to call that slot in composition (if it didn't, then what's the point of having it as a parameter?)

Therefore, invoking `slot` directly or passing `slot` as a `@Composable` parameter to another `@Composable` function could both count for "locations of usage", which there should be at most one of.

**Comment by alexvanyo** [comment]
A valid manipulation of a slot that doesn't cause a lifecycle issue:

```kotlin
@Composable
fun MyContent(
    flag: Boolean,
    modifier: Modifier = Modifier,
    content: @Composable () -> Unit,
) {
    val testContent = if (flag) {
        content
    } else {
        content
    }

    Box(modifier = modifier) {
        testContent()
    }
}
```

An invalid manipulation of a slot that does cause a lifecycle state issue:

```kotlin
@Composable
fun MyContent(
    flag: Boolean,
    modifier: Modifier = Modifier,
    content: @Composable () -> Unit,
) {
    val testContent = content

    Box(modifier = modifier) {
        if (flag) {
            testContent()
        } else {
            testContent()
        }
    }
}
```
